packages/arb-token-bridge-ui/src/hooks/useNetworksAndSigners.tsx

```diff
@@ -46,14 +46,22 @@ const defaultStatus =
     ? UseNetworksAndSignersStatus.NOT_CONNECTED
     : UseNetworksAndSignersStatus.LOADING
 
-type UseNetworksAndSignersLoadingOrErrorResult = {
+export type UseNetworksAndSignersLoadingOrErrorResult = {
   status: UseNetworksAndSignersLoadingOrErrorStatus
 }
 
-type UseNetworksAndSignersConnectedResult = {
+export type UseNetworksAndSignersConnectedResult = {
   status: UseNetworksAndSignersStatus.CONNECTED
-  l1: { network: L1Network; signer: JsonRpcSigner; provider: JsonRpcProvider }
-  l2: { network: L2Network; signer: JsonRpcSigner; provider: JsonRpcProvider }
+  l1: {
+    network: L1Network
+    provider: JsonRpcProvider
+    signer: Omit<JsonRpcSigner, 'provider'>
```

Comment:
I like the current approach, but another option we have is to "monkey patch" the provider here. something like `signer.connect(ourProvider)` would prob do it

the advantage would be that in places such as when we do `sdk.deposit(signer)`, we know it would still not use the wallet's provider